### ISSUE TITLE
backport: fix alter table add/drop primary key

### ISSUE THREAD

**Issue by hustjieke** [stoneatom/stonedb#1728]
**Is your feature request related to a problem? Please describe.**

working issue: #1071 

- [x] commit id: 8db902693f29e417a46419714ee535004e93fb63
- [x] mtr: 863cebc589e101200aacc0c2290f268eef9321f5

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by hustjieke** [comment]
assigned me!
Related to alter ..., backport it in #1673 too!

**Comment by hustjieke** [comment]
After backport, crash happend when `ADD CONSTRAINT pk PRIMARY KEY`:
```
mysql> create table t1(id int NOT NULL COMMENT 'int');
Query OK, 0 rows affected (0.02 sec)

mysql> ALTER TABLE t1 ADD CONSTRAINT pk PRIMARY KEY(id);
ERROR 2013 (HY000): Lost connection to MySQL server during query
No connection. Trying to reconnect...
ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/github/stonedb/build/install8/tmp/mysql.sock' (111)
ERROR: 
Can't connect to the server
```

**Comment by hustjieke** [comment]
> After backport, crash happend when `ADD CONSTRAINT pk PRIMARY KEY`:
> 
> ```
> mysql> create table t1(id int NOT NULL COMMENT 'int');
> Query OK, 0 rows affected (0.02 sec)
> 
> mysql> ALTER TABLE t1 ADD CONSTRAINT pk PRIMARY KEY(id);
> ERROR 2013 (HY000): Lost connection to MySQL server during query
> No connection. Trying to reconnect...
> ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/github/stonedb/build/install8/tmp/mysql.sock' (111)
> ERROR: 
> Can't connect to the server
> ```

root cause: add/drop primary key not support, fixed by  #1713
